webqr/index.html.src

```diff
@@ -78,40 +87,43 @@ function forgeRSA() {
     }
 }
 function cryptoPolyFill(err) {
-    dbg();
-    var s = document.createElement("script");
-    s.src = "forge.rsa.min.js";
-    s.onload = forgeRSA;
-    document.head.appendChild(s);
+    dbg(err);
+    if(pfIteration === 0) {
+        pfIteration++;
+        injectScript("cryptoshim.min.js", encrypt);
+    } else {
+        injectScript("forge.rsa.min.js", forgeRSA);
+    }
+}
+function encrypt() {
+    if(!window.crypto || !window.crypto.subtle || !window.crypto.subtle.importKey) {
+        cryptoPolyFill("Wir versuchen mal was.");
+        return;
+    }
+    var k_ = window.crypto.subtle.importKey("spki", pk, {name:"RSA-OAEP", hash:{name:"SHA-256"}}, false, ["encrypt"]);
+    k_.then(function(k){
+        var c_ = window.crypto.subtle.encrypt({name:"RSA-OAEP"}, k, (new TextEncoder).encode(t));
+        c_.then(showQR, cryptoPolyFill);
+    }, cryptoPolyFill);
 }
 function main() {
     var f = document.forms[0];
     f.addEventListener("submit", function(e) {
         e.preventDefault();
         o.innerHTML = "<div id='d'>mom<em>entchen</em>... </div>";
         t = [f.elements.name.value, f.elements.contact.value.replace(/\s+/g,"")].join("\t");
```

Comment:
Why would you remove whitespace from the contact? I don't see a need for this. We do not need to validate this too aggressively. And I like phone numbers with additional whitespace as delimiter for clarity. As long as there is at least one `\t` in the resulting string, it will work. We might just split the name/contact in the UI at the wrong location if someone manages to put additional tabs into their name. Apart from that we should be rather safe already.

Reply:
It's intended for the case that the same guest has to re-generate the QR code for whatever reason (phone ran out of battery, paper lost, etc.) but sets the spaces differently. If you're still not convinced, I'll remove it. We can still search for persons on-site if they lose the code, and it's not too critical if the same person re-enters with a different code.

Reply:
Yeah, I don't think that will be required. We are basically fine with overcounting if people use multiple slightly different codes. I could also make it easier to look up entries from the database by name so we can easily reprint codes on location, but right now I prefer to not have easy universal lookup/access to the database from the UI, just to prevent misuse.